src/parser/metrics/RealLinesOfCode.ts

```diff
@@ -16,75 +12,70 @@ let dlog: DebugLoggerFunction = debuglog("metric-gardener", (logger) => {
  * Counts the number of lines in a file, not counting for comments and empty lines.
  */
 export class RealLinesOfCode implements Metric {
-    private commentStatementsSuperSet: QueryStatementInterface[] = [];
+    private commentStatementsSet: Set<string>;
 
     /**
      * Constructs a new instance of {@link RealLinesOfCode}.
      * @param allNodeTypes List of all configured syntax node types.
      */
     constructor(allNodeTypes: ExpressionMetricMapping[]) {
-        allNodeTypes.forEach((expressionMapping) => {
-            if (
-                expressionMapping.metrics.includes(this.getName()) &&
-                expressionMapping.type === "statement"
-            ) {
-                const { expression, activated_for_languages } = expressionMapping;
-                const queryStatement = new ExpressionQueryStatement(
-                    expression,
-                    activated_for_languages
-                );
+        this.commentStatementsSet = new Set(
+            getExpressionsByCategory(allNodeTypes, this.getName(), "comment")
+        );
+    }
 
-                if (expressionMapping.category === "comment") {
-                    this.commentStatementsSuperSet.push(queryStatement);
-                }
-            }
-        });
+    /**
+     * Waling through the tree in order to find actual code lines.
+     *
+     * Uses a {@link TreeCursor} for this, as according to the
+     * {@link https://tree-sitter.github.io/tree-sitter/using-parsers#walking-trees-with-tree-cursors|Tree-sitter documentation},
+     * this is the most efficient way to traverse a syntax tree.
+     * @param cursor A {@link TreeCursor} for the syntax tree.
+     * @param sureCodeLines A set in which the line numbers of the found code lines are stored.
+     */
+    walkTree(cursor: TreeCursor, sureCodeLines: Set<number>) {
+        // This is not a comment syntax node, so assume it includes "real code".
+        if (!this.commentStatementsSet.has(cursor.currentNode.type)) {
```

Comment:
There might also be "whitespace" nodes. If those exists, we should also exclude those.

Reply:
I have traversed through some trees by now, and never encountered such a node. Not even with python, where I expected them most because of them having a semantic meaning. Seems like the tree-sitter grammars abstract them away when parsing the syntax tree.

Reply:
It seems like handling whitespace-sensitive languages with tree-sitter is a more complicated topic, as all grammars that parse them have to rely on external software to handle them (https://github.com/tree-sitter/tree-sitter/issues/219). Which might explain why they are not included in the resulting tree.

Reply:
Thank you for looking that up! That is good to know.